daemon/src/file_tracker.rs

```diff
@@ -423,95 +434,108 @@ impl FileTracker {
     pub fn read_dirty_files<'a, V: DBTransaction<'a, T>, T: lmdb::Transaction + 'a>(
         &self,
         iter_txn: &'a V,
-    ) -> Result<Vec<FileState>> {
-        let mut file_states = Vec::new();
-        {
-            let mut cursor = iter_txn.open_ro_cursor(self.tables.dirty_files)?;
-            for (key_result, value_result) in cursor.capnp_iter_start() {
-                let key = str::from_utf8(key_result).expect("Failed to parse key as utf8");
-                let value_result = value_result?;
-                let info = value_result.get_root::<dirty_file_info::Reader<'_>>()?;
-                let source_info = info.get_source_info()?;
-                file_states.push(FileState {
+    ) -> Vec<FileState> {
+        // TODO(happens): If we have any errors while looping over a dirty file, we
+        // should somehow be able to mark it for a retry. We can probably rely on
+        // the fact that since we skip them, they will still be dirty on the next attempt.
```

Comment:
I think you can remove the TODO and leave the comment :P We can indeed rely on it being reprocessed in the future since the dirty entry remains.